---
title: "ExpressionAnalysis"
author: "Alvaro Barbeira"
date: "June 28, 2016"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Preliminaries

This report will compare predicted gene expressions (built using different transcriptome models) to a measured (reference) transcriptome.

```{r paths, echo=FALSE}
# Reference data
WB_PATH <- "data/expression/Whole_Blood_Analysis.expr.txt.gz"
# Predicted data
PRS_BA_PATH <- "data/expression/PRS_PB8K_BETA_ALL_expression.txt.gz"
PRS_BB_PATH <- "data/expression/PRS_PB8K_BETA_BEST_expression.txt.gz"
PRS_ZA_PATH <- "data/expression/PRS_PB8K_ZSCORE_ALL_expression.txt.gz"
PRS_ZB_PATH <- "data/expression/PRS_PB8K_ZSCORE_BEST_expression.txt.gz"
DGN_PATH <- "data/expression/DGNWB_expression.txt.gz"
source("ExpressionUtilities.R")
```

Reference expression is loaded from:

```{r reference_expression, cache=TRUE}
REFERENCE <- load_file(WB_PATH)
```

Predicted data is loaded from:
```{r predicted_expression, cache=TRUE}
PRS_BA <- load_file(PRS_BA_PATH)
PRS_BA$tag = "PRS_PB8K_BETA_ALL"

PRS_BB <- load_file(PRS_BB_PATH)
PRS_BB$tag = "PRS_PB8K_BETA_BEST"

PRS_ZA <- load_file(PRS_ZA_PATH)
PRS_ZA$tag = "PRS_PB8K_ZSCORE_ALL"

PRS_ZB <- load_file(PRS_ZB_PATH)
PRS_ZB$tag = "PRS_PB8K_ZSCORE_BEST"

DGN <- load_file(DGN_PATH)
DGN$tag = "EN_DGN"

predicted <- list(DGN, PRS_BA, PRS_BB, PRS_ZA, PRS_ZB)
```

We will build a linear regression of predicted expression over reference expression with the next function.

```{r comparison_method}
comparison_results <- function(predicted, expected) {
  d1 <- predicted
  d2 <- expected
  
  d1 <- d1[rownames(d1) %in% rownames(d2), colnames(d1) %in% colnames(d2)]
  d1 <- d1[order(rownames(d1)),]

  d2 <- d2[rownames(d2) %in% rownames(d1), colnames(d2) %in% colnames(d1)]
  d2 <- d2[order(rownames(d2)),]  

  results <- build_results(d1, d2)
  results$tag <- paste0(d1$tag, "_vs_", d2$tag)
  results
}
```

And we'll be using some standard issue R tools:

```{r, message=F, warning=F}
library(ggplot2)
```


## Expression analysis

We'll gather all predicted expressions and run their comparison to the reference expression.

```{r comparison_run, cache=TRUE}
WB_DGN_COMPARISON <- comparison_results(DGN, REFERENCE)
WB_PRS_BA_COMPARISON <- comparison_results(PRS_BA, REFERENCE)
WB_PRS_BB_COMPARISON <- comparison_results(PRS_BB, REFERENCE)
WB_PRS_ZA_COMPARISON <- comparison_results(PRS_ZA, REFERENCE)
WB_PRS_ZB_COMPARISON <- comparison_results(PRS_ZB, REFERENCE)
```

And then, we'll build a plot comparing the pearson coefficients among them.
```{r r2_comparison}
build_comparison_plot_data <- function(d1, d2, facet) {
  d1 <- d1[rownames(d1) %in% rownames(d2), ]
  d1 <- d1[order(rownames(d1)),]

  d2 <- d2[rownames(d2) %in% rownames(d1), ]
  d2 <- d2[order(rownames(d2)),]
  
  result <-  data.frame(x = d1$R, y = d2$R, the_facet=facet)
  result
}
```


Lets build some plots displaying these Pearsons. **y** axis is Predicted **DGN**+ gene expression, **x** axis is predicted **PRS** gene expression,
gathered by PRS model type([BETA/ZSCORE][BEST/ALL])


```{r plot, cache=TRUE, fig.width = 5, fig.height = 20, echo=FALSE}
plot_data <- data.frame( x=numeric(), y=numeric(), the_facet=character(), x_label=character(), y_label = character())
plot_data <- rbind(plot_data, build_comparison_plot_data(WB_PRS_BA_COMPARISON, WB_DGN_COMPARISON, "DGN vs PB8K-BETA-ALL"))
plot_data <- rbind(plot_data, build_comparison_plot_data(WB_PRS_BB_COMPARISON, WB_DGN_COMPARISON, "DGN vs PB8K-BETA-BEST"))
plot_data <- rbind(plot_data, build_comparison_plot_data(WB_PRS_ZA_COMPARISON, WB_DGN_COMPARISON, "DGN vs PB8K-ZSCORE-ALL"))
plot_data <- rbind(plot_data, build_comparison_plot_data(WB_PRS_ZB_COMPARISON, WB_DGN_COMPARISON, "DGN vs PB8K-ZSCORE-BEST"))

p<-ggplot(plot_data, aes(x=x,y=y))+
            geom_point() + 
            facet_wrap(~the_facet,scales="fixed",ncol=1) +
            ggtitle("Gene Expression")

p2<- p +
    geom_abline(intercept=0, slope=1) +
      xlab("PRS PB8K") +
      ylab("DGN-WB") +
      theme_bw(20)
p2
```

## Metaxcan Results Analysis

Lets compare Metaxcan results when using the *DGN* model versus the *PRS* model.

```{r load_metaxcan_dgn, cache = TRUE, fig.width = 5, fig.height = 20, echo=FALSE}
N_ROWS = 15
DGN_FOLDER <- "data/metaxcan/dgn"
dgn_files <- list.files(DGN_FOLDER)
dgn_results <- list()
for (file in dgn_files) {
  path <- paste0(DGN_FOLDER, "/", file)
  contents <- read.csv(path, nrows=N_ROWS)
  contents <- contents[complete.cases(contents),]
  contents <- contents[!is.nan(contents$zscore),]
  contents <- contents[!is.infinite(contents$zscore),]
  key <- strsplit(file, "_DGN")[[1]][1]
  dgn_results[[key]] <- contents
}

PRS_FOLDER <- "data/metaxcan/prs_pb8k"
prs_files <- list.files(PRS_FOLDER)
metaxcan_plot_data <- data.frame( x=numeric(), y=numeric(), the_facet=character())
for (prs_file in prs_files) {
  path <- paste0(PRS_FOLDER, "/", prs_file)

  contents <- read.csv(path, nrows=N_ROWS)
  contents <- contents[!is.nan(contents$zscore),]
  contents <- contents[!is.infinite(contents$zscore),]
  if (length(contents$zscore) == 0) {
      print(paste0("Skip ", path))
      next; 
  }
  
  key <- strsplit(prs_file, "_PRS_PB8K_")[[1]][1]
  the_facet <- strsplit(prs_file, "_PRS_PB8K_")[[1]][2]
  the_facet <- strsplit(the_facet, ".csv")[[1]][1]
  
  x_avg <- mean(abs(contents$zscore))

  dgn_content <- dgn_results[[key]]
  if (length(dgn_content$zscore) == 0) {
    print(paste0("dgn skip ", key))
    next;
  }
  y_avg <- mean(abs(dgn_content$zscore))

  new_row = data.frame(x = x_avg, y = y_avg, the_facet = the_facet)
  metaxcan_plot_data <- rbind(metaxcan_plot_data, new_row)
}

p<-ggplot(metaxcan_plot_data, aes(x=x,y=y))+
            geom_point() +
            facet_wrap(~the_facet,scales="fixed",ncol=1) +
            ggtitle("Metaxcan Association") +
            annotate("text", x = 8, y = 20, label = "(Phenotype Top Result Average)")

p2<- p +
    geom_abline(intercept=0, slope=1) +
      xlab("PRS PB8k") +
      ylab("DGN-WB") +
      theme_bw(20)
p2
```

## Future Work

The naive average doesn't take into account that for different models, different genes are picked up.

